### PR TITLE
WT-13619 Move session alter api macro call to top level

### PR DESCRIPTION
**Pull request by zunyiliu**
The current session API macro call only wraps `__session_alter_internal`, when we exit `__session_alter_internal`, `API_RET_NOTFOUND_MAP` will clean the session information. Because of that `__session_blocking_checkpoint` between the two `__session_alter_internal` is called without any session name information, which causes ambiguous `CHECKPOINT_PROGRESS` messages. We should move the API macro call to the top level so it can wrap everything in `session_alter`.

### CODE REVIEW

File: src/session/session_api.c [hunk 1]
```diff
@@ -931,20 +923,28 @@ __session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
...
+    cfg[0] = cfg[1];
+    cfg[1] = NULL;
 
     /*
      * Alter table can return EBUSY error when the table is modified in parallel by eviction. Retry
      * the command after performing a system wide checkpoint. Only retry once to avoid potentially
      * waiting forever.
      */
-    ret = __session_alter_internal(session, uri, config);
+    ret = __session_alter_internal(session, uri, cfg);
```
Comment (ajmorton) [inline src/session/session_api.c:939]: I'd recommend
```suggestion
    WT_ERR_ERROR_OK(__session_alter_internal(session, uri, cfg), EBUSY, true);
```
here so we explicitly jump to the error label on a non-`EBUSY` error. It's functionally identical to the current code but adds a bit of future proofing on the off chance we add new code later.

**Review by ajmorton — APPROVED** [review]
LGTM with one suggestion 👍

File: src/session/session_api.c (hunk 1 above)
Comment (zunyiliu) [reply src/session/session_api.c]: Thank you Andrew!

**Review by sueloverso — APPROVED** [review]
LGTM

**Comment by sueloverso** [comment]
@zunyiliu the changes look good to me but please check the PR failures to see if they're related to the change.

**Pull request merged**